File: client/src/features/users/dto/response/user.response.ts
export type UserStatus = 'ACTIVE' | 'INACTIVE'

export interface RoleBriefResponse {
  id: string
  name: string
}

export interface UserResponse {
  id: string
  email: string
  displayName: string
  status: UserStatus
  tenantAdmin: boolean
  roles: RoleBriefResponse[]
  createdAt: string
  updatedAt: string
  lastLoginAt: string | null
}

export interface CreateUserResponse extends UserResponse {
  temporaryPassword: string
}

File: backend/sentinel-api/src/main/java/com/sentinel/api/user/dto/response/UserResponse.java
package com.sentinel.api.user.dto.response;

import com.sentinel.api.role.dto.response.RoleBriefResponse;
import com.sentinel.common.postgresql.user.entity.UserStatus;

import java.time.Instant;
import java.util.List;

public record UserResponse(
    String id,
    String email,
    String displayName,
    UserStatus status,
    boolean tenantAdmin,
    List<RoleBriefResponse> roles,
    Instant createdAt,
    Instant updatedAt,
    Instant lastLoginAt) {
}
